blockchain/contracts/WavGame.sol

```diff
@@ -0,0 +1,270 @@
+
+    uint256 public entryFee; // In wei
+    uint256 constant internal ENTRY_LEVEL = 1; 
+    IWavContract public wavContract; 
+    address payable public revenueContract; // Platform revenue spliting contract
+
+    mapping(address => IWavGame.Level[]) internal wavGames; // Levels are 0-indexed. i.e level 1 = 0 in level
+    mapping(address => mapping(uint256 => EnumerableSet.AddressSet)) collectors;// Collectors per level per owner game
+    mapping(address => mapping(uint256 => EnumerableSet.UintSet)) burnableSet;// Burnable IDs per level per owner game
+    mapping(address => mapping(uint256 => EnumerableSet.UintSet)) mintableSet;// Mintable IDs per level per owner game
+
+    
+    event Collected(address caller, address to, uint256 amount, uint256 refunded, uint256 totalMinted);
+    event LeveledUp(address caller, address collector, uint256 nextLevel, uint256 totalMinted);
+    event SpecialMint(address caller, address collector, uint256 id, uint256 amount);
+    event PaymentForwarded(address to, uint256 amount);
+    event GameSet(address owner, uint256 level);
+    event LevelSet(address owner, uint256 level);
+
+	// Create all initial artists
+	// create and map the 6 artist level / island to each artist, and define the Levels specific details per artist
+    constructor(IWavContract _wavContract, address _trustedForwarder,uint256 _entryFee, address payable _revenueContract) {
+        wavContract = _wavContract;
+        entryFee = _entryFee;
+        revenueContract = _revenueContract;
+        _setTrustedForwarder(_trustedForwarder);
+    }
+
+    // constructor(IWavContract _wavContract, uint256 _entryFee, address payable _revenueContract) {
+    //     wavContract = _wavContract;
+    //     entryFee = _entryFee;
+    //     revenueContract = _revenueContract;
+    // }
+
+
+    /// @notice This function mints level 1 NFTs to the recipient
+    /// @param _recipient The recipient of the minted NFT
+    /// @param _input ids and their corresponding quantities to mint
+    /// @dev This function does not support gasless, has to be called directly by user, since it requires payment in ETH and refund is possible
+    function collect(address _gameOwner, address _recipient, IDParam[] calldata _input) external override payable nonReentrant {
+        require(msg.value > 0, "PAYMENT_REQUIRED");
+        require(Helper.getLevelIndex(ENTRY_LEVEL) < wavGames[_gameOwner].length, "LEVEL_DOES_NOT_EXIST");
+
+        uint totalPayable = 0;
+        uint totalMinted = 0;
+        uint[] memory mintableIds = new uint[](_input.length); 
+        uint[] memory mintableAmountPerId = new uint[](_input.length);
+
+        // Create mintable IDs and valid quantity based on mintable set for base and msg.value
+        for (uint i; i < _input.length;) { 
+            require(mintableSet[_gameOwner][ENTRY_LEVEL].contains(_input[i].id), "NFTs_NOT_MINTABLE_FOR_LEVEL");
+            uint256 mintCost = _input[i].amount * entryFee;
```

Comment:
Can we verify that `_input[i].amount` is `>0`? Is there a way to verify `_input[i].id`?

Reply:
Great catch on _input[i].amount, will update to add that check.
The require statement on line 69 is verifying that the id sent is valid for that level.